sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs

```diff
@@ -661,27 +667,15 @@ async Task<Result> GetScopeProperties (MessageId msg_id, int scope_id, Cancellat
 
 				var values = res.Value? ["result"]? ["value"]?.Values<JObject> ().ToArray ();
 
-				if(values == null)
+				if(values == null || values.Length == 0)
 					return Result.OkFromObject (new { result = Array.Empty<object> () });
 
-				var var_list = new List<object> ();
-				int i = 0;
-				for (; i < vars.Length && i < values.Length; i ++) {
-					// For async methods, we get locals with names, unlike non-async methods
-					// and the order may not match the var_ids, so, use the names that they
-					// come with
-					if (values [i]["name"] != null)
-						continue;
-
-					ctx.LocalsCache[vars [i].Name] = values [i];
-					var_list.Add (new { name = vars [i].Name, value = values [i]["value"] });
-				}
-				for (; i < values.Length; i ++) {
-					ctx.LocalsCache[values [i]["name"].ToString()] = values [i];
-					var_list.Add (values [i]);
+				foreach (var value in values) {
+					var name = value ["name"]?.Value<string> ();
+					ctx.LocalsCache [value ["name"]?.ToString ()] = value;
```

Comment:
why do you fetch the value using .Value```<>``` and then discard it before using ToString?